test/smoke/testApps/SimpleEventData/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SimpleEventDataTest.java

```diff
@@ -0,0 +1,56 @@
+package com.microsoft.applicationinsights.smoketest;
+
+import java.util.concurrent.TimeUnit;
+
+import com.microsoft.applicationinsights.internal.schemav2.Data;
+import com.microsoft.applicationinsights.internal.schemav2.Envelope;
+import com.microsoft.applicationinsights.internal.schemav2.EventData;
+
+import org.junit.*;
+
+import static org.junit.Assert.*;
+
+public class SimpleEventDataTest extends AiSmokeTest {
+
+	@Test
+	public void doCalcSendsRequestDataAndEventData() throws Exception {
```

Comment:
Seems like a single test method has multiple asserts. Can we break this down into multiple tests? Would help in making more fine grained analysis and better maintenance.

Reply:
You mean that we should not have multiple asserts for a single test method ? We need to add another arthritic test method ?

Reply:
@aivendortest I think that yes a single test method should have 1 assert statement or at max 2 asserts. This particular test methods seems to have more than 5 asserts and single assert failing means whole tests fails and we cannot debug each smaller pieces independently then,

Reply:
I disagree, @dhaval24. A test can have multiple asserts. It should encapsulate a "logical test case," e.g. if you're testing a form that creates a new customer, you should assert on each field of the new customer object.